lib/queue.rb

```diff
@@ -1,31 +1,95 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = []
+    @front = -1
+    @rear = -1
+    @size = 20
   end
 
   def enqueue(element)
```

Comment:
This _mostly_ works, but you need to check when the Queue is full and you add an element. In that case front and back shouldn't be -1 and `(rear + 1) % length` would be equal to front.